### PR TITLE
Blogpost

### PR DESCRIPTION
**Pull request by choffmeister**

### CODE REVIEW

File: _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+layout: post
+title: "Code splitting and tree shaking with Typescript and Webpack"
+date: "2017-10-26 14:41:02"
+image: blog-header/tekom17.jpg
+categories: tech-stories
+authors: ["Christian H."]
+---
+
+As off today single page applications are all around and bring application-like user experience to the browser. If done naively, one problem is that the JavaScript for the application gets very big which is not only a problem in terms of network transfer, but also the client browser have to parse a lot of JavaScript upfront before being able to make the application interactive. This is especially a problem for mobile devices. To reduce initial loading and parsing of unneeded JavaScript there is [code splitting][code-splitting]. To drop unused code all together there is [tree shaking][tree-shaking]. This blog post will show you how to get these two working with your [TypeScript][typescript]+[Webpack][webpack] project in a step by step guide.
```
Comment (BirgitBader) [inline _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md]: What do you mean by "as off today"?

File: _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    "target": "esnext",
+    // produce a source map
+    "sourceMap": true
+  }
+}
+{% endhighlight %}
+
+## Babel configuration
+
+Now you will get JavaScript that uses a bunch of features, that are not widely supported by now. With the preset `babel-preset-env` you can define not what exactly [Babel][babel] transform you want to apply, but what browsers you want to support and then `babel-preset-env` picks the right configuration to match your needs. Your `.babelrc` should look somewhat like this:
```
Comment (BirgitBader) [inline _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md]: Don't understand the second sentence. Can you rephrase, pleazzzzee?

File: _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        ],
+      },
+    ],
+  },
+  // produce sourcemap
+  devtool: '#sourcemap',
+}
+{% endhighlight %}
+
+That is it. Now you not only enabled Webpack to do code splitting and tree shaking, but you also reduced your JavaScript asset size even more, because the use of `babel-preset-env` instead of letting TypeScript do the whole compilation. For example `async`/`await` will most likely just be in your resulting JavaScript instead of being transpiled to generators, which is just way shorter.
```
Comment (BirgitBader) [inline _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md]: what is meant by "transpiled"? Couldn't find this...

File: _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md (hunk 1 above)
Comment (choffmeister) [reply _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md]: "Heutzutage"... Falsches Englisch? 😄

File: _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md (hunk 3 above)
Comment (choffmeister) [reply _posts/2017/2017-10-30-typescript-codesplitting-treeshaking.md]: https://www.stevefenton.co.uk/2012/11/compiling-vs-transpiling/

**Comment by choffmeister** [comment]
@BirgitBader Maybe this photo: https://www.pexels.com/photo/road-in-city-during-sunset-248159/

Is not really related to the post, but I don't know any picture that would make sense.
